### PR TITLE
fix(MetronomeStore): cleanup deleted jobs

### PR DESCRIPTION
**Pull request by ahoskins**
When setting the new jobs, remove the previously deleted jobs from the
`jobMap`.

**Now:**
screen recording 2018-01-04 at 05 25 pm

**Before this change:**
screen recording 2018-01-04 at 05 32 pm

**Testing**:
1. Checkout this branch
2. Create a job
3. Delete the job
4. It will now delete from the jobs table without needing a page reload

Closes DCOS-20118

**Checklist**
- [ ] Did you add a JIRA issue in a commit message or as part of the branch name?
- [ ] Did you add new unit tests?
- [ ] Did you add new integration tests?
- [ ] If this is a regression, did you write a test to catch this in the future?

### CODE REVIEW

**Comment by DanielMSchmidt** [comment]
Alfred test this please 😏

File: src/js/stores/MetronomeStore.js [hunk 1]
```diff
@@ -165,6 +165,7 @@ class MetronomeStore extends EventEmitter {
           this.emit(METRONOME_JOB_SCHEDULE_UPDATE_SUCCESS, action.jobID);
           break;
         case REQUEST_METRONOME_JOBS_SUCCESS:
+          this.removeOldJobs(action.data);
```
Comment (DanielMSchmidt) [inline src/js/stores/MetronomeStore.js]: Can we add a small test for this?

**Comment by DanielMSchmidt** [comment]
I don't know if it's flackiness or not, but on my machine I got 4 errors on the integration tests on master, but 14 on this branch. Just FYI 👍

**Comment by ahoskins** [comment]
@DanielMSchmidt thanks, these are legit failures, I will fix.

**Comment by ahoskins** [comment]
Alfred test this please

**Comment by DanielMSchmidt** [comment]
Awesome, down to 4 errors locally, will retrigger alfred 👍

**Comment by DanielMSchmidt** [comment]
Alfred test this please

File: src/js/stores/MetronomeStore.js (hunk 1 above)
Comment (ahoskins) [reply src/js/stores/MetronomeStore.js]: Yes, good idea, I'll add a test in `test-jobs.js` (our end to end system tests) 👍

**Comment by ahoskins** [comment]
@DanielMSchmidt I've added a system test to prevent this fix from regressing. We've disabled the system test for now on Github, you can run it locally with:

```
docker run -it --rm --ipc=host \
    -v `pwd`:/dcos-ui \
    -e CLUSTER_URL=<url-of-live-cluster-from-ccm> \
    mesosphere/dcos-ui dcos-system-test-driver --only "jobs" -v \
    ./system-tests/driver-config/cluster.sh
```

**Comment by DanielMSchmidt** [comment]
Alfred test this please 😏

**Comment by ahoskins** [comment]
alfred test this please

**Pull request merged**